### PR TITLE
Bug/ion list and accessibility

### PR DESCRIPTION
**Pull request by szutrifork**
ion-list implemented

### CODE REVIEW

File: src/kirby/components/list/list.component.html [hunk 1]
```diff
@@ -1,61 +1,58 @@
...
-        <ng-container
-          *ngTemplateOutlet="
-            sectionHeaderTemplate;
-            context: { $implicit: section.name }
-          ">
-        </ng-container>
-      </li>
-      <li *ngFor="let item of section.items; let idx = index;" [class.last-in-section]="idx === section.items.length - 1"
+
+<ion-list mode="ios" kirbyInfiniteScroll (scrollEnd)="onLoadOnDemand()" [disabled]="!isLoadOnDemandEnabled">
```
Comment (jakobe) [inline src/kirby/components/list/list.component.html]: Replace with `ion-infinite-scrol`: https://ionicframework.com/docs/api/infinite-scroll

File: src/kirby/components/list/list.component.html [hunk 2]
```diff
@@ -1,61 +1,58 @@
...
+  <ion-list-header class="list-header" *ngIf="listHeaderTemplate">
+    <ng-container *ngTemplateOutlet="listHeaderTemplate"></ng-container>
+  </ion-list-header>
+  
+  <div *ngIf="isSectionsEnabled; else normalList">
```
Comment (jakobe) [inline src/kirby/components/list/list.component.html]: Why this additional div?

File: src/kirby/components/list/list.component.html [hunk 3]
```diff
@@ -1,61 +1,58 @@
...
+    <div *ngFor="let section of groupedItems">
```
Comment (jakobe) [inline src/kirby/components/list/list.component.html]: Why this additional div?

File: src/kirby/components/list/list.component.html [hunk 4]
```diff
@@ -1,61 +1,58 @@
...
+        <ion-list-header class="section-header">
+          <ion-label class="kirby-text-small">{{ section.name }}</ion-label>
```
Comment (jakobe) [inline src/kirby/components/list/list.component.html]: Consider moving to css:
`ion-list-header.section-header > ion-label {...}`

```suggestion
          <ion-label>{{ section.name }}</ion-label>
```

Comment (jakobe) [inline src/kirby/components/list/list.component.html]: :fire: You have removed the ability for a sectionHeaderTemplate???

File: src/kirby/components/list/list.component.html [hunk 5]
```diff
@@ -1,61 +1,58 @@
...
-      <li *ngFor="let item of section.items; let idx = index;" [class.last-in-section]="idx === section.items.length - 1"
+
+<ion-list mode="ios" kirbyInfiniteScroll (scrollEnd)="onLoadOnDemand()" [disabled]="!isLoadOnDemandEnabled">
+  <ion-list-header class="list-header" *ngIf="listHeaderTemplate">
+    <ng-container *ngTemplateOutlet="listHeaderTemplate"></ng-container>
+  </ion-list-header>
+  
+  <div *ngIf="isSectionsEnabled; else normalList">
+    <div *ngFor="let section of groupedItems">
+        <ion-list-header class="section-header">
```
Comment (jakobe) [inline src/kirby/components/list/list.component.html]: You should use `ion-item-divider`: https://ionicframework.com/docs/api/item-divider

File: src/kirby/components/list/list.component.html [hunk 6]
```diff
@@ -1,61 +1,58 @@
...
-          *ngTemplateOutlet="
-            sectionHeaderTemplate;
-            context: { $implicit: section.name }
-          ">
-        </ng-container>
-      </li>
-      <li *ngFor="let item of section.items; let idx = index;" [class.last-in-section]="idx === section.items.length - 1"
+
+<ion-list mode="ios" kirbyInfiniteScroll (scrollEnd)="onLoadOnDemand()" [disabled]="!isLoadOnDemandEnabled">
+  <ion-list-header class="list-header" *ngIf="listHeaderTemplate">
```
Comment (jakobe) [inline src/kirby/components/list/list.component.html]: Just use a tag selector:
```suggestion
  <ion-list-header *ngIf="listHeaderTemplate">
```
CSS:
`ion-list-header {...}`

File: src/kirby/components/list/list.component.scss [hunk 7]
```diff
@@ -10,80 +11,39 @@ $shadow-mask: 0 -6px 0px 0px get-color('white');
...
-
-ul {
-  list-style-type: none;
-  padding-left: 0;
-  margin: 0;
-  font-size: font-size('s');
-
-  li {
-    padding: size('m');
+  .list-ios,
```
Comment (jakobe) [inline src/kirby/components/list/list.component.scss]: This class is not known to us - you are using inner implementation details from Ionic...

File: src/kirby/components/list/list.component.spec.ts [hunk 8]
```diff
@@ -64,50 +75,39 @@ describe('ListComponent', () => {
...
         items: Item.createItems(1, 2, 3),
       });
       runNgOnChanges();
 
-      const liElements = spectator.queryAll('li');
+      const liElements = spectator.queryAll('ion-item');
       expect(liElements.length).toEqual(component.items.length);
     });
-
-    it('should render one li element for each item and one for each section, if sections are enabled', () => {
```
Comment (jakobe) [inline src/kirby/components/list/list.component.spec.ts]: Why has this test been deleted?

File: src/kirby/components/list/list.component.ts [hunk 9]
```diff
@@ -57,7 +57,16 @@ export class ListComponent implements OnChanges {
    */
   @HostBinding('class.rounded')
   @Input()
```
Comment (jakobe) [inline src/kirby/components/list/list.component.ts]: This is not an input

**Comment by labanos** [comment]
closes #256

File: src/kirby/components/list/list.component.html (hunk 1 above)
Comment (hafstad) [reply src/kirby/components/list/list.component.html]: Due to missing page-object (ion-content, ion-app), we will implement ion-inifinite-scroll in another pull-req

**Pull request merged**